src/runtime/vulkan_memory.h

```diff
@@ -161,13 +161,9 @@ int VulkanMemoryAllocator::destroy(void *user_context, VulkanMemoryAllocator *in
         error(user_context) << "VulkanBlockAllocator: Unable to destroy instance! Invalide instance pointer!\n";
         return halide_error_code_internal_error;
     }
-    const BlockAllocator::MemoryAllocators &allocators = instance->block_allocator->current_allocators();
+    BlockAllocator::MemoryAllocators allocators = instance->block_allocator->current_allocators();
     instance->destroy(user_context);
-    BlockAllocator::destroy(user_context, instance->block_allocator);
-    if (allocators.system.deallocate == nullptr) {
-        error(user_context) << "VulkanBlockAllocator: Unable to destroy instance! Missing system allocator interface!\n";
-        return halide_error_code_internal_error;
-    }
+    halide_abort_if_false(user_context, allocators.system.deallocate != nullptr);
```

Comment:
Is this really a crash-the-program error?

Reply:
Yeah, probably safer to revert back to the error message and return code.